packages/babel-plugin-transform-es2015-destructuring/test/fixtures/destructuring/es7-object-rest/expected.js

```diff
@@ -3,7 +3,12 @@ var x = babelHelpers.objectWithoutProperties(z, []);
 var x = z.x;
 var y = babelHelpers.objectWithoutProperties(z, ["x"]);
 var x = z[x];
-var y = babelHelpers.objectWithoutProperties(z, [x]);
+var y = babelHelpers.objectWithoutProperties(z, [typeof x === "symbol" ? x : `${ x }`]);
```

Comment:
I thought it would avoid the runtime cost for non-computed keys, but I guess it's better to reimplement it that way.

Can also fix the iteration to work on own property names/symbols while at it.